### PR TITLE
Add validations to forms

### PR DESCRIPTION
**Pull request by vsavkin**

### CODE REVIEW

**Comment by mhevery** [comment]
I need to better understand this. Let's talk about it in person tomorrow.

File: modules/angular2/src/facade/collection.dart [hunk 1]
```diff
@@ -67,6 +67,14 @@ class StringMapWrapper {
   static void forEach(Map m, fn(v, k)) {
     m.forEach((k, v) => fn(v, k));
   }
+  static HashMap merge(HashMap a, HashMap b) {
+    var m = {};
```
Comment (vicb) [inline modules/angular2/src/facade/collection.dart]: would it be faster ?

```
var m = new HashMap.from(a);
b.forEach((k, v) => m[k] = v);
return m;
```

Comment (vsavkin) [inline modules/angular2/src/facade/collection.dart]: I'll fix it

**Comment by vsavkin** [comment]
Merged via https://github.com/angular/angular/commit/ded83e589bf72bfd43fdb44c7623a382e13fee19

**Pull request closed without merging**